### PR TITLE
Introduce Contribution Doc

### PR DESCRIPTION
**Pull request by awgreene**

### CODE REVIEW

File: CONTRIBUTING.md
Comment (everettraven) [inline CONTRIBUTING.md:72]: Do PRs actually require reviews from another subproject's maintainers right now?
```suggestion
Currently, PRs required at least one approval from a operator-controller maintainer in order to get merged.
```

File: CONTRIBUTING.md
Comment (everettraven) [inline CONTRIBUTING.md:59]: Something that might be nice to add here is some guidelines on when and how to reach out to maintainers directly about PRs. For example, is it expected that my PR may take 2 weeks to get a review? If I haven't received a review in a week, where & who should I reach out to to make sure my changes are getting reviewed?

File: CONTRIBUTING.md
Comment (everettraven) [inline CONTRIBUTING.md:76]:
```suggestion
The coding style suggested by the Golang community is used throughout the operator-controller project:
```

File: CONTRIBUTING.md
Comment (everettraven) [inline CONTRIBUTING.md:81]:
```suggestion
In addition to the linked style documentation, operator-controller formats Golang packages using the `golangci-lint` tool. Before
```

File: CONTRIBUTING.md
Comment (everettraven) [inline CONTRIBUTING.md:85]:
```suggestion
Please follow this style to make the operator-controller project easier to review, maintain and develop.
```

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:59]: Good point! We may want to add an OWNERS or MAINTAINERS file and then simply mention to tag the individuals in that group.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Ah, I had started this file by copying RukPak's CONTRIBUTION.md, I had meant to swap all RukPak references with operator-controller. I will address this in all places.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:59]: I've created [an issue](https://github.com/operator-framework/operator-controller/issues/153) to track this work

File: CONTRIBUTING.md
Comment (dtfranz) [inline CONTRIBUTING.md:72]: Very small fix:
```suggestion
Currently, PRs require at least one approval from a operator-controller maintainer in order to get merged.
```

Comment (varshaprasad96) [inline CONTRIBUTING.md:72]: Can we add a section mentioning requirements to open a PR. Something like [this](https://sdk.operatorframework.io/docs/contribution-guidelines/opening-pull-requests/#pr-checklist). Since this repository is in a fairly new state, we could ask for PR description, tests, relevant documentation as some tips for opening a PR.

File: CONTRIBUTING.md
Comment (varshaprasad96) [inline CONTRIBUTING.md:13]: +1 ! Nice addition to have the details handy :)

File: CONTRIBUTING.md
Comment (ankitathomas) [inline CONTRIBUTING.md:36]: should this link to a subsection?

File: CONTRIBUTING.md
Comment (ankitathomas) [inline CONTRIBUTING.md:37]: We possibly need to mention test/coverage expectations with codecov.

File: CONTRIBUTING.md
Comment (tmshort) [reply CONTRIBUTING.md:19]: Git hates trailing whitespace

File: CONTRIBUTING.md
Comment (ankitathomas) [inline CONTRIBUTING.md:6]: Do we require contributors to agree to the CNCF contributor license agreement?

File: CONTRIBUTING.md
Comment (tmshort) [reply CONTRIBUTING.md:72]: That's info that could be added to the PR template.

File: CONTRIBUTING.md
Comment (ankitathomas) [inline CONTRIBUTING.md:21]: There is a more detailed getting started section in README.md, so we may want to consider moving this section there.

File: CONTRIBUTING.md
Comment (ankitathomas) [inline CONTRIBUTING.md:28]: Should we have some guidelines on how to create and tag issues for features, bugs or security vulnerabilities?

File: CONTRIBUTING.md
Comment (ankitathomas) [inline CONTRIBUTING.md:39]: Outlining the expected response from the maintainers might be useful, so contributors know what to expect.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:72]: I think this is covered by [the existing PR template](https://github.com/operator-framework/operator-controller/blob/main/.github/workflows/pull_request_template.md).

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:36]: Great idea!

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:37]: I don't think that we have any hardcore code coverage expectations, but I'll call out the need for tests.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:6]: Right now we don't enforce it, could be a good thing to bring up at the WG meeting.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:39]: Good idea, I'll add a section asking them to tag us if we don't respond within a week.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:21]: Done.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:72]: @varshaprasad96 please mark this as resolved if you agree.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:20]: Might be a better use of our time to create a few [issue templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository).

Comment (awgreene) [reply CONTRIBUTING.md:20]: Captured this idea in an issue: https://github.com/operator-framework/operator-controller/issues/154

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Addressed.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Added the spaces as I would prefer a newline.

File: CONTRIBUTING.md
Comment (tmshort) [inline CONTRIBUTING.md:17]: Although we mention trivial fixes don't need a separate issue; where's the dividing line? A PR itself could be a bug report and fix in one, and having to create an issue (which, unrelatedly, share the same number spaces as PRs) seems like duplication of effort.

File: CONTRIBUTING.md
Comment (tmshort) [inline CONTRIBUTING.md:85]: Link? (we have links to all other stuff, perhaps here too?)

File: CONTRIBUTING.md
Comment (tmshort) [inline CONTRIBUTING.md:1]: Do we want to include:

1. The build process?
2. The test process?
3. Required tools for building and testing (e.g. `kind`).

Comment (tmshort) [reply CONTRIBUTING.md:1]: If we use taskfiles (https://github.com/operator-framework/enhancements/issues/126), this will potentially become important. Alternatively, if this ought to be part of the README, then I'd reference that earlier.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md:66]: Good question, I added similar links and descriptions earlier in the doc, perhaps making that a section and linking it here is the right idea?

Comment (everettraven) [reply CONTRIBUTING.md:66]: Ah I see, I don't think it hurts to be extra verbose in case that section is missed or someone is going through and looking at all the links and forgets that information was already provided. I think there is quite a bit of information to digest when it comes to OLMv1 so being that bit more verbose and having *some* repetition might be beneficial

File: CONTRIBUTING.md
Comment (ankitathomas) [reply CONTRIBUTING.md:1]: The README should be the first thing anyone will see when they get to the repo, so it makes more sense to me to have setup, build and test should be called out either there.

File: CONTRIBUTING.md
Comment (dtfranz) [inline CONTRIBUTING.md:5]: Minor grammatical/phrasing change:
```suggestion
Thank you for your interest in contributing to the Operator-Controller. 
```

File: CONTRIBUTING.md
Comment (dtfranz) [inline CONTRIBUTING.md:12]: Tiny fix and ending period for consistency:
```suggestion
- [The Deppy project](https://github.com/operator-framework/deppy/), which enables the operator-controller to identify valid installs and upgrades from the list of installable content provided via the Catalogd project.
```

File: CONTRIBUTING.md
Comment (dtfranz) [inline CONTRIBUTING.md:24]: Mostly a preference nit:
```suggestion
2. The community creates GitHub Issues, GitHub Discussions, or pull requests in the appropriate repos based on (1) to continue the discussion.
```

File: CONTRIBUTING.md
Comment (dtfranz) [inline CONTRIBUTING.md:95]: I don't think this make target is actually in the project yet :sweat_smile:

Comment (awgreene) [reply CONTRIBUTING.md:95]: Oh, let's add that as a follow up issue.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Addressed in the latest commit.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Addressed in the latest commit.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Addressed in the latest commit.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Addressed in the latest commit.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Addressed in the latest commit.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Addressed in the latest commit.

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Addressed in the latest commit.

**Review by dtfranz — APPROVED** [review]
lgtm!

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: I've created [an issue](https://github.com/operator-framework/operator-controller/issues/160) to add some Architecture docs

File: CONTRIBUTING.md
Comment (awgreene) [reply CONTRIBUTING.md]: Linked

**Pull request merged**
